### ISSUE TITLE
A lint suggesting `.get(n)` instead of `if n < slice.len() { slice[n] } ...`

### ISSUE THREAD

**Issue by nyurik** [rust-lang/rust-clippy#16166]
### What it does

When recently reviewing new Rust dev code, I saw this pattern, which is a clear dup of the `Slice::get`. It would be great to catch it.  Note that in an ideal case, we shouldn't catch this specific case, but instead catch any kind of stdlib function patterns (ASTs), and suggest replacements (this might generate some false positives though)

### Advantage

- shorter, more readable code
- clearer intent

### Example

```rust
fn get_item(index: usize) -> Option<&'static str> {
  if index < GLOBAL_ARRAY.len() {
    Some(GLOBAL_ARRAY[index])
  } else {
    None
  }
}
```

Could be written as:

```rust
fn get_item(index: usize) -> Option<&'static str> {
  GLOBAL_ARRAY.get(index)
}
```

**Comment by andjsrk** [comment]
It seems that `indexing_slicing` in restriction group covers this: [playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=55f7d8c04045e50314beec4c9b1e753d)

**Comment by nyurik** [comment]
@andjsrk thx, TIL, and I will enable it for many of my mission critical projects. That said, I think this is different. The proposed lint is to catch a manual implementation of a built-in `.get(...)` function rather a blanket ban of a slice access.  We already do it in many lints that work with Results and Option enums.

**Comment by andjsrk** [comment]
~~I guess you tested it with arrays. Note that the lint is type-aware: [playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=aaa85a93cf744e663938b4e8e3fcefd9)~~

EDIT: oh I misread your comment, sorry
